### PR TITLE
Checkstyle and CodeQL

### PR DESCRIPTION
**Pull request by marcgs**
## What this PR changes/adds

- Run checkstyle on PRs in both upstream and forks.
- Run CodeQL on  PRs in both upstream and forks.
- Fix CodeQL findings

## Why it does that

Both Checkstyle and CodeQL are already in use in the EDC framework, so these tools are the natural choice in related projects like the Identity Hub as well.

## Linked Issue(s)

Relates to #254

## Checklist

- [x] added appropriate tests?
- [x] performed checkstyle check locally?
- [x] added/updated copyright headers?
- [x] documented public classes/methods?
- [x] added/updated relevant documentation?
- [x] added relevant details to the changelog? (_skip with label `no-changelog`_)
- [x] formatted title correctly? (_take a look at the [CONTRIBUTING](https://github.com/eclipse-dataspaceconnector/identityservice/blob/main/CONTRIBUTING.md#submit-a-pull-request) and [styleguide](https://github.com/eclipse-dataspaceconnector/identityservice/blob/main/styleguide.md) for details_)

### CODE REVIEW

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      - 'dependabot/**'
+  pull_request:
+    branches: [ main ]
+    paths-ignore:
+      - '**.md'
+      - 'docs/**'
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
```
Comment (ouphi) [inline .github/workflows/codeql-analysis.yml:13]: Why is this needed?

File: .github/workflows/codeql-analysis.yml [hunk 2]
```diff
@@ -0,0 +1,50 @@
+name: "CodeQL"
+
+on:
+  push:
+    branches-ignore:
+      - 'dependabot/**'
```
Comment (ouphi) [inline .github/workflows/codeql-analysis.yml]: What does it do?

File: extensions/identity-hub-verifier/src/main/java/org/eclipse/dataspaceconnector/identityhub/verifier/DidJwtCredentialsVerifier.java [hunk 3]
```diff
@@ -60,6 +61,7 @@ public boolean isSignedByIssuer(SignedJWT jwt) {
         return signatureResult.succeeded();
     }
 
+    @Override
```
Comment (ouphi) [inline extensions/identity-hub-verifier/src/main/java/org/eclipse/dataspaceconnector/identityhub/verifier/DidJwtCredentialsVerifier.java]: nit: You can merge main to avoid spurious diff as this have been already fixed in the last identityhub commit in main.

File: .github/workflows/codeql-analysis.yml (hunk 1 above)
Comment (marcgs) [reply .github/workflows/codeql-analysis.yml:13]: To prevent having multiple CodeQL runs on the same branch. Only the last run is relevant, so we cancel the other ones.

File: .github/workflows/codeql-analysis.yml (hunk 2 above)
Comment (marcgs) [reply .github/workflows/codeql-analysis.yml]: I think it was added in the EDC repo to prevent runs on that branch. I don't think it is relevant for this repo, so I have removed it.

File: .github/workflows/codeql-analysis.yml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: [ 'java' ]
+        # CodeQL supports [ 'cpp', 'csharp', 'go', 'java', 'javascript', 'python' ]
```
Comment (algattik) [inline .github/workflows/codeql-analysis.yml]: These 3 comment lines seem redundant, I would delete them

File: .github/workflows/verify.yaml [hunk 5]
```diff
@@ -2,12 +2,26 @@ name: Test Code (Style, Tests)
...
     paths-ignore:
       - '**.md'
       - 'docs/**'
 
 jobs:
+  Checkstyle:
+    permissions:
+      id-token: write
+      checks: write
+      pull-requests: write
```
Comment (algattik) [inline .github/workflows/verify.yaml]: why are these permissions needed?

**Review by algattik — CHANGES_REQUESTED** [review]

File: .github/workflows/verify.yaml (hunk 5 above)
Comment (marcgs) [reply .github/workflows/verify.yaml]: It's the same set of permissions as granted in EDC. I assume "pull-request" is required to show Checkstyle errors in the PR? But I am not sure... I will try to remove them and see what happens.

Comment (algattik) [reply .github/workflows/verify.yaml]: Seems more like a leftover from using a PR action previously

Comment (algattik) [reply .github/workflows/verify.yaml]: Note this https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/pull/1651

Comment (cpeeyush) [reply .github/workflows/verify.yaml]: checks and pull-request write permission is needed by unit test result publishing action as per the official doc https://github.com/EnricoMi/publish-unit-test-result-action#permissions.

So, this permission can be moved where we have this action(EnricoMi/publish-unit-test-result-action).

Not sure why It needs this permission because we still see the test result report as PR comment but it is needed as per official doc, so I am highlighting it here.

Comment (marcgs) [reply .github/workflows/verify.yaml]: I have added these permissions to the "Test" job for correctness, although it has been working without them anyways.

**Review by cpeeyush — APPROVED** [review]
Approved with suggestions.

File: .github/workflows/codeql-analysis.yml [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: [ 'java' ]
```
Comment (pkirch) [inline .github/workflows/codeql-analysis.yml:23]: Why a matrix for one element?

**Review by pkirch — APPROVED** [review]

File: .github/workflows/codeql-analysis.yml (hunk 6 above)
Comment (marcgs) [reply .github/workflows/codeql-analysis.yml:23]: This is the template generated by Github when enabling CodeQL. It is a matrix so that it is easy to add additional languages. Although not required in our case I think it does not hurt either, and I would leave it for consistency with EDC.

File: .github/workflows/codeql-analysis.yml [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+
+    steps:
+      - uses: actions/checkout@v3
```
Comment (cpeeyush) [inline .github/workflows/codeql-analysis.yml:26]: nit: here we use v3 but in other workflows we use v2. Just a small suggestion is if looks okay then we can keep it consistent to use v2.

File: .github/workflows/codeql-analysis.yml [hunk 8]
```diff
@@ -0,0 +1,45 @@
+name: "CodeQL"
+
+on:
+  push:
+  pull_request:
+    branches: [ main ]
+    paths-ignore:
+      - '**.md'
+      - 'docs/**'
```
Comment (cpeeyush) [inline .github/workflows/codeql-analysis.yml:9]: nit: Maybe we do it this way:

```
  push:
    branches:
      - main
    paths-ignore:
      - 'docs/**'
  pull_request:
    paths-ignore:
      - 'docs/**'
```

then it will run on PR against all branch and not just main but push will be restricted to main branch only.

I took this sample from MVD.So we can do similar here.

**Review by ouphi — APPROVED** [review]
